Assets/Scripts/Basic Player Movement/CharacterController.cs

```diff
@@ -0,0 +1,56 @@
+using UnityEngine;
+
+public class CharacterController : MonoBehaviour
+{
+    [SerializeField]
+    private float speed = 6f; // Speed of the character
+    [SerializeField]
+    private float jumpForce = 5f; // Force of the character's jump
+    [SerializeField]
+    private float rotationSpeed = 80f; // Rotation speed of the character
+    [SerializeField]
+    private Transform groundCheck; // Transform used to check if the character is grounded
+
+    private Rigidbody rb; // Rigidbody component of the character
+    private bool isGrounded; // Flag to track if the character is grounded
+
+    void Start()
+    {
+        // Get the Rigidbody component
+        rb = GetComponent<Rigidbody>();
+    }
+
+    void Update()
+    {
+        // Check if the character is grounded
+        isGrounded = Physics.CheckSphere(groundCheck.position, 0.1f);
+
+        // Get the horizontal and vertical input from the keyboard
+        float horizontalInput = Input.GetAxis("LookHorizontal");
+        float horizontalMoveInput = Input.GetAxis("Horizontal");
+        float verticalInput = Input.GetAxis("Vertical");
+
+
+        // Move the character based on the input
+        transform.position -= transform.forward * verticalInput * speed * Time.deltaTime;
+        transform.position += transform.right * horizontalMoveInput * speed * Time.deltaTime;
+        transform.Rotate(Vector3.up, horizontalInput * rotationSpeed * Time.deltaTime);
+
+        // Jump if the character is grounded and the space bar is pressed
+        if (isGrounded && Input.GetButtonDown("Jump"))
+        {
+            rb.AddForce(Vector3.up * jumpForce, ForceMode.Impulse);
+        }
+
+    }
+
+    void OnCollisionEnter(Collision collision)
+    {
+        // Check if the character has collided with an object marked as "Ground"
+        if (collision.gameObject.tag == "Ground")
```

Comment:
```suggestion
        if (collision.gameObject.CompareTag("Ground"))
```
Compare tag is the way your "supposed" to check for tags. Don't know why, possibly more efficient?

Edit: My AI friend has the following to say
![image](https://user-images.githubusercontent.com/33167757/208777883-f9488332-40b9-4b55-b493-37ccd170c292.png)